### ISSUE TITLE
Does it support SFT training?

### ISSUE THREAD

**Issue by Lomax314** [jzhang38/EasyContext#38]
I noticed that the code does not support the passing of attention_mask, making it impossible to perform padding operations for SFT data？

https://github.com/jzhang38/EasyContext/blob/fe494926fd770c28762e5bd514d96f13f2297f86/easy_context/zigzag_ring_attn/monkey_patch.py#L26

In addition, will there be any issues with the loss calculation method in the code for SFT data where the labels contain values of -100 (the prompt and padding parts)?

https://github.com/jzhang38/EasyContext/blob/fe494926fd770c28762e5bd514d96f13f2297f86/train.py#L117-L138

Look forward to your response. Thank you.
